### PR TITLE
Added Dark Theme Implementation

### PR DESCRIPTION
**Pull request by AdityaP700**
With reference to the issue number #37 
Added Dark theme across the page with the changes being :
- Updated background and text colors across all pages.
- Seamless transition between light and dark modes.
- Optimized elements such as buttons, links, and images for better readability and aesthetics in dark mode.
- set the Local Storage to store the response

https://github.com/user-attachments/assets/087d2425-6437-4e7f-b525-f3b7f2560b6b

## Release Notes

- **New Features**
  - Introduced a `ThemeProvider` for managing light and dark themes across the application.
  - Added dark mode styling to multiple components, enhancing visual adaptability for users.
  - Implemented a theme toggle feature in the Navbar and Register components.

- **Bug Fixes**
  - Improved text visibility and background colors in dark mode for various components.

- **Documentation**
  - Updated context and theme management documentation for better clarity on usage.

These changes enhance user experience by providing a cohesive and responsive interface that adapts to user preferences for light or dark themes.

### CODE REVIEW

**Comment by RamakrushnaBiswal** [comment]
hi @AdityaP700 you can use darkmode.js

**Comment by AdityaP700** [comment]
Hello @RamakrushnaBiswal ,Well do you mean to create or replace a file named themeContext.js with darkmode.js file as i have written logic for the dark_mode theme in the themeContext.js and applied changes of the theme individually using the logic .?

File: src/Components/Pages/Register.jsx [hunk 1]
```diff
@@ -4,16 +4,29 @@ import pic3 from "../../assets/img/abt2.png";
...
+    document.body.classList.toggle("dark");
+    const currentTheme = document.body.classList.contains("dark") ? "dark" : "light";
+    localStorage.setItem("theme", currentTheme);
+  };
+  React.useEffect(() => {
+    const savedTheme = localStorage.getItem("theme");
+    if (savedTheme === "dark") {
+      document.body.classList.add("dark");
+    }
+  }, []);
```
[the remark this answers is not in the document]
Comment (AdityaP700) [reply src/Components/Pages/Register.jsx:19]: Alright will look into it

**Comment by RamakrushnaBiswal** [comment]
@AdityaP700 no . I mean use it https://darkmodejs.learn.uno/

**Comment by AdityaP700** [comment]
> @AdityaP700 no . I mean use it https://darkmodejs.learn.uno/
Alright will work upon it and give the needful result soon.

**Comment by RamakrushnaBiswal** [comment]
@AdityaP700 did you work upon it

**Comment by AdityaP700** [comment]
> @AdityaP700 did you work upon it
Well Hi @RamakrushnaBiswal ,well Apology for the delay . I reviewed and tested it but i didn't find it appealing much,thus i have issue while customizing it thus i decided to hold it for now .

**Comment by RamakrushnaBiswal** [comment]
@AdityaP700 ??

**Comment by AdityaP700** [comment]
Screenshot 2024-10-11 212625
Screenshot 2024-10-11 221556
Hey @RamakrushnaBiswal, I am quite Close to implement the entire feature of darkmode.js but still got some minor issues ,as in while switching to the dark mode, the navbar font do not change but got the logic for the one  for scrolling and all . So that's how it's going

**Pull request closed without merging**